### ISSUE TITLE
.glb using KHR-MC as external shape

### ISSUE THREAD

**Issue by andreasplesch** [x3dom/x3dom#758]
do not render. See https://rawgit.com/andreasplesch/gltf-test/Binary-MC/index.html .

minimal example: https://rawgit.com/andreasplesch/gltf-test/Binary-MC/examples/x3dom/index.html?category=sampleModels&model=Box&scale=1&type=glTF-Binary-MaterialsCommon

and
https://github.com/cx20/gltf-test/issues/17
KhronosGroup/glTF-Sample-Models#109

This is surprising since a lot of effort and code is there to support KHR MC materials. Unfortunately, the CesiumMilkTruck .glb in the example does not use the KHR MC extension but internal shaders.

Is there a test .glb with KHR MC materials which was used during development ? There may be only a minor issue.

As a workaround it is possible to use gltf-pipeline to convert from KHR materials to generic gltf material.

**Comment by andreasplesch** [comment]
I spend some time trying to track this down but was not successful. The debug (press 'd') render works, eg. the position and normal buffers are available. Adding an additional light does not help. The correct shader is loaded.

**Comment by npolys** [comment]
Greetings from SIGGRAPH!
Web3d showed well and our team of members really rocked it  all week! :-)

The gltf 2.0 is coming like a freightrain... lots of excitement here in
LA...

Thanks for looking into this.... its important that we can inline a whole
file and eventually can selectively use nodes through the url.glb#id
 mechanism
(Eg transform, shape, material, appearance,....)
Great stuff!
This is  win for the x3d ecosystem


On Friday, August 4, 2017, Andreas Plesch <notifications@github.com> wrote:
> I spend some time trying to track this down but was not successful. The
debug (press 'd') render works, eg. the position and normal buffers are
available. Adding an additional light does not help. The correct shader is
loaded.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci5.googleusercontent.com/proxy/z_vsBC0e5sctowMP8VvPbIJDvAKFRFnA3dHlQ0s_XDdnPM8ii3D69tfQYkHrev-ioQSd1SymZ0GCCY38zTLJKavCKbQ1pzazYr-DOYaepCKFoGq_X404DtkLwCvU4xH0b6kDdKlOC96GesND8PJJelXX86wZ_w=s0-d-e1-ft#https://github.com/notifications/beacon/AHU5M2RoNATOBZGo5WLZtT5N2SNehFm8ks5sU7DEgaJpZM4OuAAe.gif
>

-- 
Nicholas F. Polys, Ph.D.

Director of Visual Computing
Virginia Tech Research Computing

Affiliate Professor
Virginia Tech Department of Computer Science

**Comment by andreasplesch** [comment]
Thanks for the SIGGRAPH note.  With the focus on glTF 2.0, time may be better spend on a 2.0 loader for x3dom. SIGGRAPH may be a good opportunity to gather input from the x3dom community on plans for glTF 2.0.

**Comment by andreasplesch** [comment]
I had a suspicion and are making some progress:
https://rawgit.com/andreasplesch/gltf-test/c44de016c0a20e617a2cc1ea3d97e6cedc6f927f/index.html
https://rawgit.com/andreasplesch/gltf-test/c44de016c0a20e617a2cc1ea3d97e6cedc6f927f/examples/x3dom/index.html?category=sampleModels&model=Duck&scale=1&type=glTF-Binary-MaterialsCommon
Replacing the material parsed from the file with a default (black) MC material works.

**Comment by andreasplesch** [comment]
I think I found it:
https://rawgit.com/andreasplesch/gltf-test/b8e5c82cad821f8c3e9b20314033ad2b2b269f5f/index.html
https://rawgit.com/andreasplesch/gltf-test/b8e5c82cad821f8c3e9b20314033ad2b2b269f5f/examples/x3dom/index.html?category=sampleModels&model=Duck&scale=1&type=glTF-Binary-MaterialsCommon

The sample models all have a transparency value of 1.0 which is supposed to be interpreted as an _opacity_, confusingly but according to spec.
However, x3dom interpreted this as _transparency_ and made all models completely see-through. Above I patched the transparency value so the x3dom shader sees it as opacity.
The real fix will be to modify the shader which will be a very small fix.

aopt may be an issue if it currently produces non-compliant glb with material commons. I think it would need to be fixed as well.

I will test the shader fix and produce a small PR for consideration.
